lib/components/Box.js

```diff
@@ -0,0 +1,106 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+
+/*
+ * general box component that can render children
+ * vertical mode:
+ *   renders a single row with each child in a column
+ * horizontal mode:
+ *   renders each child in a row
+ * merges any prop styles into styles object
+ */
+class Box extends PureComponent {
+  static get propTypes() {
+    return {
+      axis: PropTypes.oneOf(['vertical', 'horizontal']),
+      children: PropTypes.oneOfType([
+        PropTypes.node,
+        PropTypes.arrayOf(PropTypes.node),
+      ]),
+      justify: PropTypes.string,
+      onClick: PropTypes.func,
+      style: PropTypes.object,
+      paddingTop: PropTypes.string,
+      margin: PropTypes.string,
+    };
+  }
+
+  static get defaultProps() {
+    return {
+      onClick: () => undefined,
+      paddingTop: '0rem',
+      margin: 'auto',
+      style: {},
+    };
+  }
+
+  definedOrNull(object) {
+    return object || object === null;
+  }
+
+  /*
+   * merges any style props into the style object
+   * also merges in null values so that defaults can
+   * be removed
+   */
+  reduceStyle() {
+    const { paddingTop, margin } = this.props;
+    let { style } = this.props;
+
+    if (this.definedOrNull(paddingTop)) style.paddingTop = paddingTop;
+
+    if (this.definedOrNull(margin)) style.margin = margin;
+
+    return style;
+  }
+
+  render() {
+    const { axis, children, justify, onClick } = this.props;
+
+    const style = this.reduceStyle();
+
+    if (axis === 'vertical') {
+      /*
+       * render each child into a column
+       * NOTE: don't pass too many children
+       */
+
+      let len;
+      if (Array.isArray(children)) len = children.length;
+      else if (children) len = 1;
+      else len = 12;
+
+      const colSize = `col-md-${12 / len}`;
```

Comment:
what if there a number of children not a factor of 12?

Reply:
why does it need to be a factor of 12? it just has to be less than 12
https://github.com/twbs/bootstrap/blob/5543229950051c2a2b9c1dc49a7d548759adc3d6/dist/css/bootstrap-grid.css#L84

Reply:
The class doesn’t need to be, but the way I read this line, if you have 5 items for example won’t this turn the class into `col-md-2.5` which wouldn’t work.

Reply:
I pushed up a fix to the `box` branch